server-spi-private/src/main/java/org/keycloak/storage/ExportImportManager.java

```diff
@@ -13,4 +14,6 @@ public interface ExportImportManager {
     void importRealm(RealmRepresentation rep, RealmModel newRealm, boolean skipUserDependent);
 
     void updateRealm(RealmRepresentation rep, RealmModel realm);
+
+    UserModel createUser(RealmModel realm, UserRepresentation userRep);
```

Comment:
Disclaimer: For the time being, it's ok as is, but will need fixing before merging to main.

Currently the `*Representation` is taken from the common module. The representation will depend on the storage module. Hence the API will need to be revamped in terms of `String` / `InputStream` parameters rather than `*Representation`.

Reply:
@hmlnarik - I wonder if export/import could be a separate PR. What else would need to be part of this PR? The "extract the logic from the services module into a separate maven module so that services would only refer the abstract layer" from the original issue doesn't reveal much to me.

Reply:
I believe it could. What needs to be done for "this (upstream) PR" is to remove all reference to user storage SPI / federated users from the services / server-spi / server-spi-private, and redirect all non-core calls in `DefaultKeycloakSession` for an area provider to the core one, meaning e.g. `userLocalStorage()` body would be rewritten to `return users();`